pkg/controller/syncsetinstance/syncsetinstance_controller.go

```diff
@@ -667,28 +643,39 @@ func (r *ReconcileSyncSetInstance) applySyncSetResources(ssi *hivev1.SyncSetInst
 	case hivev1.CreateOnlySyncSetApplyBehavior:
 		applyFn = h.Create
 	}
-
+	// determine if we can gather info for the resource and apply it
 	var applyErr error
-	for i, resource := range resources {
-		resourceSyncStatus := hivev1.SyncStatus{
-			APIVersion: infos[i].APIVersion,
-			Kind:       infos[i].Kind,
-			Resource:   infos[i].Resource,
-			Name:       infos[i].Name,
-			Namespace:  infos[i].Namespace,
-			Hash:       r.hash(resource.Raw),
-		}
+	var discoveryErr error
```

Comment:
We do not need a separate variable for `discoveryErr`. We can use `applyErr` in its place.